app/handlers/api.py

```diff
@@ -45,6 +45,108 @@
 from os.path import splitext
 
 
+class AutoCropperHandler(BaseHandler):
+    SUPPORTED_METHODS = ("GET", "POST")
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def get(self,):
+        response = yield Task(
+            self.api_call,
+            url=self.settings['API_URL'] + '/autocropper',
+            method='GET')
+        self.set_json_output()
+        if response.code == 404:
+            body = loads(response.body.decode("utf-8"))
+            self.finish(self.json_encode({
+                        'status': 'success',
+                        'message': body['message'],
+                        'data': []}))
+        else:
+            self.set_status(response.code)
+            self.finish(response.body)
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def post(self):
+
+
+        if self.request.files:
+            fileinfo = self.request.files['file'][0]
+            body = fileinfo['body']
+            fname = fileinfo['filename']
+            fileencoded = base64.b64encode(body).decode('utf-8')
+
+            body = {
+                "filename": fname,
+                "image": fileencoded,
+                'content_type': fileinfo['content_type']
+            }
+
+            response = yield Task(
+                self.api_call,
+                url=self.settings['API_URL'] + '/autocropper',
+                method='POST',
+                body=self.json_encode(body))
+            self.set_json_output()
+            self.set_status(response.code)
+            self.finish(response.body)
+
+
+class AutoCropperUploaderHandler(BaseHandler):
+    SUPPORTED_METHODS = ("POST",)
+
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def post(self):
+
+
+        if self.request.files:
+            fileinfo = self.request.files['file'][0]
+            body = fileinfo['body']
+            fname = fileinfo['filename']
+            info(fname)
+            dirfs = dirname(realpath(__file__))
+            info(dirfs)
+            fh = open(dirfs + '/' + fname, 'wb')
```

Comment:
A nice to have would be refactoring out some of the logic in both post methods in AutoCropperHandler and AutoCropperUploaderHandler since a lot of values in the request body are similar